pkg/clients/cloudsql/cloudsql.go

```diff
@@ -283,10 +296,17 @@ func LateInitializeSpec(spec *v1beta1.CloudSQLInstanceParameters, in sqladmin.Da
 
 // IsUpToDate checks whether current state is up-to-date compared to the given
 // set of parameters.
-func IsUpToDate(in *v1beta1.CloudSQLInstanceParameters, currentState sqladmin.DatabaseInstance) bool {
-	currentParams := &v1beta1.CloudSQLInstanceParameters{}
-	LateInitializeSpec(currentParams, currentState)
-	return cmp.Equal(in, currentParams, cmpopts.IgnoreInterfaces(struct{ resource.AttributeReferencer }{}))
+func IsUpToDate(name string, in *v1beta1.CloudSQLInstanceParameters, currentState *sqladmin.DatabaseInstance) (bool, error) {
+	genState, err := copystructure.Copy(currentState)
+	if err != nil {
+		return false, err
+	}
+	_, ok := genState.(*sqladmin.DatabaseInstance)
```

Comment:
Another downside of doing generic deep copy is the necessity of type casting :/

Reply:
You might be able to avoid this by using a different deepcopy library, for example https://github.com/jinzhu/copier or https://github.com/ulule/deepcopier. I'd prefer we used a library that copied into a pointer rather than returning `interface{}`, if possible.

Reply:
I actually tried out the first you mention and was running into issues with it not performing a full deep copy so pointer values were causing the copied from struct to be modified. I’ll give the second a try though!

Reply:
@negz unfortunately both of the libraries mentioned (and all others that do not copy into `interface{}` that I can find) copy pointers by reference